### PR TITLE
Change extra hosts

### PR DESCRIPTION
**Pull request by wild-endeavor**
# TL;DR
I think this used to work because the flyte binary used to run as a process inside the top level container.  Now it runs in the cluster.  In order for K8s to call out the the webhook service (which in the `--dev` case is running on the user's host machine), we need to pipe through the host machine's IP.

This works in conjunction with https://github.com/flyteorg/flyte/pull/3228.

This has been tested with both the `--dev` switch and without it.

## Type
- [x] Bug Fix
- [ ] Feature
- [ ] Plugin

## Are all requirements met?

- [x] Code completed
- [x] Smoke tested
- [ ] Unit tests added
- [ ] Code documentation added
- [ ] Any pending items have an associated Issue

## Complete description
The extra hosts was first added in https://github.com/flyteorg/flytectl/pull/369.  At the time the flyte binary ran as a process in the dind container.  I guess at that time, K8s was able to see the dind container as 127.0.0.1, and so routing there would hit the flyte binary.

### CODE REVIEW

**Review by jeevb — APPROVED** [review]
🙌

**Review by pingsutw — APPROVED** [review]
